src/nacl/bindings/crypto_kx.py

```diff
@@ -0,0 +1,150 @@
+               'Seed must be a {0} byte long bytes sequence'.format(
+                   crypto_kx_SEED_BYTES),
+               raising=exc.TypeError)
+        res = lib.crypto_kx_seed_keypair(public_key, secret_key, seed)
+    ensure(res == 0, "Key generation failed.", raising=exc.CryptoError)
+
+    return (ffi.buffer(public_key, crypto_kx_PUBLIC_KEY_BYTES)[:],
+            ffi.buffer(secret_key, crypto_kx_SECRET_KEY_BYTES)[:])
+
+
+def crypto_kx_client_session_keys(client_public_key,
+                                  client_secret_key,
+                                  server_public_key):
+    """
+    Generate session keys for the client.
+    :param client_public_key:
+    :type client_public_key: bytes
+    :param client_secret_key:
+    :type client_secret_key: bytes
+    :param server_public_key:
+    :type server_public_key: bytes
+    :return: (decryption_key, encryption_key)
+    :rtype: (bytes, bytes)
+    """
+    ensure(isinstance(client_public_key, bytes) and
+           len(client_public_key) == crypto_kx_PUBLIC_KEY_BYTES,
+           'Client public key must be a {0} bytes long bytes sequence'.format(
+               crypto_kx_PUBLIC_KEY_BYTES),
+           raising=exc.TypeError)
+    ensure(isinstance(client_secret_key, bytes) and
+           len(client_secret_key) == crypto_kx_SECRET_KEY_BYTES,
+           'Client secret key must be a {0} bytes long bytes sequence'.format(
+               crypto_kx_PUBLIC_KEY_BYTES),
+           raising=exc.TypeError)
+    ensure(isinstance(server_public_key, bytes) and
+           len(server_public_key) == crypto_kx_PUBLIC_KEY_BYTES,
+           'Server public key must be a {0} bytes long bytes sequence'.format(
+               crypto_kx_PUBLIC_KEY_BYTES),
+           raising=exc.TypeError)
+
+    decryption_key = ffi.new("unsigned char[]", crypto_kx_SESSION_KEY_BYTES)
+    encryption_key = ffi.new("unsigned char[]", crypto_kx_SESSION_KEY_BYTES)
+    res = lib.crypto_kx_client_session_keys(decryption_key,
+                                            encryption_key,
+                                            client_public_key,
+                                            client_secret_key,
+                                            server_public_key)
+    ensure(res == 0,
+           "Client session key generation failed.",
+           raising=exc.CryptoError)
+
+    return (ffi.buffer(decryption_key, crypto_kx_SESSION_KEY_BYTES)[:],
+            ffi.buffer(encryption_key, crypto_kx_SESSION_KEY_BYTES)[:])
+
+
+def crypto_kx_server_session_keys(server_public_key,
+                                  server_secret_key,
+                                  client_public_key):
+    """
+    Generate session keys for the server.
+    :param server_public_key:
+    :type server_public_key: bytes
+    :param server_secret_key:
+    :type server_secret_key: bytes
+    :param client_public_key:
+    :type client_public_key: bytes
+    :return: (decryption_key, encryption_key)
```

Comment:
I don't like the renaming of the shared key directions `rx` and `tx` to `decryption` and `encryption`, and i fear it could be misleading for downstream users; better keep with upstream choices `rx_key` and `tx_key`